plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -73,41 +74,58 @@ class KeycloakError(Exception):
     pass
 
 
-def get_token(base_url, validate_certs, auth_realm, client_id,
-              auth_username, auth_password, client_secret):
-    if not base_url.lower().startswith(('http', 'https')):
+def get_token(module_params):
+    """ Obtains connection header with token for the authentication,
+        token already given or obtained from credentials
+        :param module_params: parameters of the module
+        :return: connection header
+    """
+    token = module_params.get('token')
+    base_url = module_params.get('auth_keycloak_url')
+
+    if not None and not base_url.lower().startswith(('http', 'https')):
```

Comment:
```suggestion
    if base_url not None and not base_url.lower().startswith(('http', 'https')):
```

Reply:
Though I would say that `base_url` should always be there, so better would be:
```suggestion
    if not base_url.lower().startswith(('http', 'https')):
```

Reply:
Yes sorry my bad, I thought it was only needed for the token generation